types/src/chain_id.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use anyhow::{format_err, Error, Result};
+use serde::{de::Visitor, Deserialize, Deserializer, Serialize};
+use std::{
+    convert::TryFrom,
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+
+/// A registry of reserved chain IDs
+/// Its main purpose is to improve human readability of reserved chain IDs in config files and CLI
+/// When signing transactions for such chains, the numerical chain ID should still be used
+/// (e.g. MAINNET has numeric chain ID 0, PREMAINNET has chain ID 1, etc)
+#[repr(u8)]
+#[derive(Copy, Clone, Debug)]
+pub enum ReservedChain {
+    MAINNET,
+    PREMAINNET,
+    TESTNET,
+    DEVNET,
+    TESTING,
+}
```

Comment:
When it comes to reserved Libra networks, I agree MAINNET = 0.

However, PREMAINNET, TESTNET, DEVNET, TESTING, are non-reserved in my opinion.  They could evolve over time.  One thing we could do if we want to keep a single enum is something like below - renaming to NamedChain and then clearly laying out what is reserved and what is not:

```suggestion
pub enum NamedChain {
    /// MAINNET is the Libra mainnet production chain and is reserved for 0.
    MAINNET,
    // The below chains are non-reserved, non-production and may change over time.  They are listed for convenience here.
    PREMAINNET,
    TESTNET,
    DEVNET,
    TESTING,
}
```